swarm/storage/localstore.go

```diff
@@ -184,3 +184,40 @@ func (ls *LocalStore) Iterator(from uint64, to uint64, po uint8, f func(Address,
 func (ls *LocalStore) Close() {
 	ls.DbStore.Close()
 }
+
+func (ls *LocalStore) Migrate() error {
+	schema, err := ls.DbStore.GetSchema()
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+
+	log.Debug("found schema", "schema", schema, "runtime-schema", CurrentDbSchema)
+	if schema != CurrentDbSchema {
+		// run migrations
+
+		if schema == "" {
+			log.Debug("running migrations for", "schema", schema, "runtime-schema", CurrentDbSchema)
+
+			cleanupFunc := func(c *chunk) bool {
+				valid := false
+				for _, v := range ls.Validators {
+					if valid = v.Validate(c.Address(), c.Data()); valid {
+						break
+					}
+				}
+				return valid
+			}
+
+			ls.DbStore.Cleanup(cleanupFunc)
+
+			err := ls.DbStore.PutSchema(DbSchemaPurity)
+			if err != nil {
+				log.Error(err.Error())
```

Comment:
do we need to log this here too? wouldn't this be logged by callee of `NewSwarm()`?

Reply:
not sure it will be - better log twice than not even once.